diff -ruN -X Excl a/system/core/fingerprintd/FingerprintDaemonProxy.cpp b/system/core/fingerprintd/FingerprintDaemonProxy.cpp
--- a/system/core/fingerprintd/FingerprintDaemonProxy.cpp	2016-09-06 02:17:24.914107961 +0300
+++ b/system/core/fingerprintd/FingerprintDaemonProxy.cpp	2016-10-05 12:32:49.000000000 +0300
@@ -23,6 +23,7 @@
 #include <keystore/IKeystoreService.h>
 #include <keystore/keystore.h> // for error codes
 #include <utils/Log.h>
+#include <sys/stat.h>
 
 #include "FingerprintDaemonProxy.h"
 
@@ -180,7 +181,16 @@
     ALOG(LOG_VERBOSE, LOG_TAG, "nativeOpenHal()\n");
     int err;
     const hw_module_t *hw_module = NULL;
-    if (0 != (err = hw_get_module(FINGERPRINT_HARDWARE_MODULE_ID, &hw_module))) {
+    struct stat st;
+    if (stat("/sys/class/goodix_fp", &st)) {
+        err = hw_get_module(FINGERPRINT_HARDWARE_MODULE_ID, &hw_module);
+        ALOGI("Try open fingerprint HW Module %i\n", err);
+    }
+    else {
+        err = hw_get_module_by_class(FINGERPRINT_HARDWARE_MODULE_ID, "goodix", &hw_module);
+        ALOGI("Try open fingerprint.goodix HW Module %i\n", err);
+    }
+    if (0 != err) {
         ALOGE("Can't open fingerprint HW Module, error: %d", err);
         return 0;
     }
